src/compiler/scala/tools/nsc/typechecker/MethodSynthesis.scala

```diff
@@ -305,17 +297,26 @@ trait MethodSynthesis {
       /* Explicit isSetter required for bean setters (beanSetterSym.isSetter is false) */
       final def completer(sym: Symbol) = namerOf(sym).accessorTypeCompleter(tree, isSetter)
       final def fieldSelection         = Select(This(enclClass), basisSym)
-      final def derivedMods: Modifiers = mods & flagsMask | flagsExtra mapAnnotations (_ => Nil)
 
       def derivedSym: Symbol = tree.symbol
       def derivedTree: Tree  = EmptyTree
 
+      // (Ultimately,) there will be no field for these vals:
+      //   - abstract vals have no value we could store (until they become concrete, potentially)
+      //   - [Emitted, later removed during AddInterfaces/Mixins] concrete vals in traits can't have a field
+      //   - [Emitted, later removed during Constructors] a concrete val with a statically known value (Unit / ConstantType)
+      //     performs its side effect according to lazy/strict semantics, but doesn't need to store its value
+      //     each access will "evaluate" the RHS (a literal) again
+      // We would like to avoid emitting unneccessary fields, but the required knowledge isn't available until after typer.
+      // The only way to avoid emitting & suppressing, is to not emit at all until we are sure to need the field, as dotty does.
```

Comment:
> but the required knowledge isn't available until after typer.

Is it available *during* typer, at the point that we are adding the derived field to a template? If so, at that juncture, we could omit the field and unlink the speculatively created field symbol from the class scope.

Given the initial creation of the derived field is already split between namer (`enterGetterSetter`) and typer (